lib/bundler/fetcher/dependency_fetcher.rb

```diff
@@ -0,0 +1,75 @@
+require 'bundler/fetcher/fetcher_impl'
+
+module Bundler
+  class Fetcher
+    class DependencyFetcher < FetcherImpl
```

Comment:
This is the level of true nits, but I think the constant names would be slightly improved by `Fetcher::FetcherImpl` → `Fetcher::Base` or `Fetcher::Shared` and `DependencyFetcher` → `Fetcher::Dependency`.